ert/data/record/_record.py

```diff
@@ -241,27 +319,53 @@ def _flatten_record_dict(
 
     @property
     def data(self) -> record_data:
-        # internally, recordtree is represented as BlobRecord; we don't need data here
+        """Internally, RecordTree is represented as BlobRecord; therefore
+        this property shoudn't be used.
```

Comment:
"The RecordTree is a BlobRecords with data containing URIs for its leaf node records, but this implementation is to be regarded as private"